[libpsl] static library linked to dynamic libunistring.dll.a

### Description / Steps to reproduce the issue

Hello,

It seems that `libpsl.a` from mingw-w64-x86_64-libpsl 0.21.2-1 is linked to a dynamic library:

```
nm -C /mingw64/lib/libpsl.a | grep -i "__imp_u"
                 U __imp_uninorm_nfkc
```

This makes static building of [Sunshine](https://github.com/LizardByte/Sunshine), which depends on libcurl, to be impossible. Would it be possible to fix this so that static builds are possible again? 

### Expected behavior

Static builds that use libpsl.a (via libcurl) should link without issue.

### Actual behavior

```
[100%] Linking CXX executable sunshine.exe
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../lib\libpsl.a(psl.c.obj):(.text+0x141b): undefined reference to `__imp_uninorm_nfkc'
D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../x86_64-w64-mingw32/bin/ld.exe: D:/a/_temp/msys64/mingw64/bin/../lib/gcc/x86_64-w64-mingw32/12.2.0/../../../../lib\libpsl.a(psl.c.obj):(.text+0x1497): undefined reference to `__imp_uninorm_nfkc'
```

### Verification

- [X] I have verified that my MSYS2 is up-to-date before submitting the report (see https://www.msys2.org/docs/updating/)

### Windows Version

MINGW64_NT-10.0-19045

### MINGW environments affected

- [X] MINGW64
- [ ] MINGW32
- [ ] UCRT64
- [ ] CLANG64
- [ ] CLANG32
- [ ] CLANGARM64

### Are you willing to submit a PR?

Yes (but I am not 100% certain how to fix this issue).

--- a/mingw-w64-libpsl/PKGBUILD
+++ b/mingw-w64-libpsl/PKGBUILD
@@ -4,7 +4,7 @@ _realname=libpsl
 pkgbase=mingw-w64-${_realname}
 pkgname=("${MINGW_PACKAGE_PREFIX}-${_realname}")
 pkgver=0.21.2
-pkgrel=1
+pkgrel=2
 pkgdesc="Public Suffix List library (mingw-w64)"
 arch=('any')
 mingw_arch=('mingw32' 'mingw64' 'ucrt64' 'clang64' 'clang32' 'clangarm64')
@@ -14,14 +14,13 @@ depends=("${MINGW_PACKAGE_PREFIX}-libiconv"
          "${MINGW_PACKAGE_PREFIX}-libidn2"
          "${MINGW_PACKAGE_PREFIX}-libunistring"
          "${MINGW_PACKAGE_PREFIX}-gettext")
-makedepends=("${MINGW_PACKAGE_PREFIX}-libxslt"
-             "${MINGW_PACKAGE_PREFIX}-python"
-             "${MINGW_PACKAGE_PREFIX}-gettext"
-             "${MINGW_PACKAGE_PREFIX}-gtk-doc"
+makedepends=("${MINGW_PACKAGE_PREFIX}-cc"
              "${MINGW_PACKAGE_PREFIX}-meson"
-             "${MINGW_PACKAGE_PREFIX}-pkg-config"
              "${MINGW_PACKAGE_PREFIX}-ninja"
-             "${MINGW_PACKAGE_PREFIX}-cc")
+             "${MINGW_PACKAGE_PREFIX}-pkg-config"
+             "${MINGW_PACKAGE_PREFIX}-python"
+             "${MINGW_PACKAGE_PREFIX}-gtk-doc"
+             "${MINGW_PACKAGE_PREFIX}-libxslt")
 source=("https://github.com/rockdaboot/libpsl/releases/download/${pkgver}/${_realname}-${pkgver}.tar.gz"
         001-Fix-build-on-Windows.patch::https://github.com/rockdaboot/libpsl/commit/c47cf796.patch)
 sha256sums=('e35991b6e17001afa2c0ca3b10c357650602b92596209b7492802f3768a6285f'
@@ -33,6 +32,21 @@ prepare() {
 }
 
 build() {
+  # static
+  [[ -d "${srcdir}/build-static-${MSYSTEM}" ]] && rm -rf "${srcdir}/build-static-${MSYSTEM}"
+  mkdir -p "${srcdir}/build-static-${MSYSTEM}" && cd "${srcdir}/build-static-${MSYSTEM}"
+
+  MSYS2_ARG_CONV_EXCL="--prefix=" \
+  ${MINGW_PREFIX}/bin/meson setup \
+    --prefix="${MINGW_PREFIX}" \
+    --buildtype plain \
+    --default-library=static \
+    -Druntime=libidn2 \
+    ../${_realname}-${pkgver}
+
+  ${MINGW_PREFIX}/bin/meson compile
+
+  # shared
   [[ -d "${srcdir}/build-${MSYSTEM}" ]] && rm -rf "${srcdir}/build-${MSYSTEM}"
   mkdir -p "${srcdir}/build-${MSYSTEM}" && cd "${srcdir}/build-${MSYSTEM}"
 
@@ -40,7 +54,7 @@ build() {
   ${MINGW_PREFIX}/bin/meson setup \
     --prefix="${MINGW_PREFIX}" \
     --buildtype plain \
-    --default-library=both \
+    --default-library=shared \
     -Druntime=libidn2 \
     ../${_realname}-${pkgver}
 
@@ -48,6 +62,8 @@ build() {
 }
 
 package() {
+  cd "${srcdir}/build-static-${MSYSTEM}"
+  DESTDIR="${pkgdir}" ${MINGW_PREFIX}/bin/meson install
   cd "${srcdir}/build-${MSYSTEM}"
   DESTDIR="${pkgdir}" ${MINGW_PREFIX}/bin/meson install
 

